TrafficCapture/captureKafkaOffloader/src/main/java/org/opensearch/migrations/trafficcapture/kafkaoffloader/KafkaCaptureFactory.java

```diff
@@ -0,0 +1,95 @@
+import org.opensearch.migrations.trafficcapture.model.CaptureOutputStreamMetadata;
+
+import java.io.ByteArrayOutputStream;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Properties;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.atomic.AtomicLong;
+
+@Slf4j
+public class KafkaCaptureFactory implements IConnectionCaptureFactory {
+
+    private static final String DEFAULT_MUTATING_TOPIC = "logging-mutating-topic";
+    private static final String DEFAULT_NON_MUTATING_TOPIC = "logging-non-mutating-topic";
+
+    private final Producer<String, byte[]> producer;
+    private final String mutatingTopicName;
+    private final String nonMutatingTopicName;
+
+    public KafkaCaptureFactory(String producerPropertiesPath, String mutatingTopicName, String nonMutatingTopicName) throws IOException {
+        Properties producerProps = new Properties();
+        try {
+            producerProps.load(new FileReader(producerPropertiesPath));
+        } catch (IOException e) {
+            System.err.println("Unable to locate provided properties file path: " + producerPropertiesPath);
+            throw e;
+        }
+        // There is likely some default timeout/retry settings we should configure here to reduce any potential blocking
+        // i.e. the Kafka cluster is unavailable
+        producer = new KafkaProducer<>(producerProps);
+        this.mutatingTopicName = mutatingTopicName;
+        this.nonMutatingTopicName = nonMutatingTopicName;
+    }
+
+    public KafkaCaptureFactory(String producerPropertiesPath) throws IOException {
+        this(producerPropertiesPath, DEFAULT_MUTATING_TOPIC, DEFAULT_NON_MUTATING_TOPIC);
+    }
+
+    @Override
+    public IChannelConnectionCaptureSerializer createOffloader(String connectionId) throws IOException {
+        AtomicLong supplierCallCounter = new AtomicLong();
+        Map<CodedOutputStream, ByteArrayOutputStream> codedStreamToByteStreamMap = new HashMap<>();
+        return new StreamChannelConnectionCaptureSerializer(connectionId,
+            () -> {
+                // Potentially enhance to use java nio ByteBuffer here if we know size beforehand
+                ByteArrayOutputStream baos = new ByteArrayOutputStream();
+                var cos = CodedOutputStream.newInstance(baos);
+                codedStreamToByteStreamMap.put(cos, baos);
+                return cos;
+            },
+            (offloaderInput) -> {
+                try {
+                    CodedOutputStream cos = offloaderInput.getCodedOutputStream();
+                    CaptureOutputStreamMetadata metadata = offloaderInput.getCaptureOutputStreamMetadata();
+                    ByteArrayOutputStream byteStream = codedStreamToByteStreamMap.get(cos);
+                    byte[] bytes = byteStream.toByteArray();
+                    byteStream.close();
```

Comment:
close() can throw an IOException, though I honestly cannot think of why it would in this case.  If it did though, everything after this would not be run, including the removal from the map.